### ISSUE TITLE
[RN 0.55.4] - Incompatibility between `react-native link` and Pods

### ISSUE THREAD

**Issue by olivierlesnicki** [facebook/react-native#19468]
- [x] I have reviewed the [documentation](https://facebook.github.io/react-native)
- [x] I have searched [existing issues](https://github.com/facebook/react-native/issues)
- [x] I am using the [latest React Native version](https://github.com/facebook/react-native/releases)

## Environment
Environment:
  OS: macOS High Sierra 10.13.1
  Node: 9.3.0
  Yarn: 1.3.2
  npm: 5.5.1
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: Not Found

Packages: (wanted => installed)
  react: 16.3.1 => 16.3.1
  react-native: 0.55.4 => 0.55.4

## Steps to Reproduce
1. Create a new react-native app
2. Setup the ios project to use Pods
3. Install react-native-spinkit
4. Run react-native link command
5. Run pod install inside the ios folder

## Expected Behavior
It shouldn't install React as part of the CocoaPod installation

## Actual Behavior
It installs React as a Pod as part of the CocoaPod installation (because react-native-spinkit refers it in its .Podspec)

**Comment by olivierlesnicki** [comment]
This was raised previously (with react-native-firebase as an example): https://github.com/facebook/react-native/issues/16912 with an older version of React Native.

It's also the topic of a further thread: https://github.com/facebook/react-native/pull/15460#issuecomment-346345788 that never went anywhere.

**Comment by olivierlesnicki** [comment]
The  temporary hack I've found to go around this issue is to delete the package's podspec after installation but before linking - ensuring it doesn't modify the .Podfile.

**Comment by vasilich6107** [comment]
@olivierlesnicki just paste it into your podfile

```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    if target.name == "react"
      target.remove_from_project
    end
  end
end
```

**Comment by vovkasm** [comment]
We follow one rule about this and did not deal with CLI bugs anymore: Don't mix pods and `react-native link` )))
With first pod, we transform the application to use full pod schema including react-native and stops using `react-native link` functionality. I think this is better way to manage ios deps, and can recommend it. See my answer to the [SO question for example](https://stackoverflow.com/a/48044426/2266462) (slightly outdated, but may be useful).
There are some other gotchas... specifically with swift and frameworks... but that's another story...

**Comment by lswank** [comment]
This is absolutely a problem still. [This](https://github.com/facebook/react-native/pull/15460#issuecomment-346345788) talks about a modified version of react-native link, and it looks like it got merged into the mainline at about 0.50, but my project is sitting at 0.55.4 and this hasn't resolved anything. Modifying the Podfile to remove react from the project after install doesn't seem to work, nor does pointing the Podfile to node_modules/.
